test/rules/best-practises/no-empty-blocks.js

```diff
@@ -123,4 +171,230 @@ describe('Linter - no-empty-blocks', () => {
     const labelIndex = Math.floor(lastItemIndex / 5) * 4
     return items[labelIndex]
   }
+
+  it('should respect config severity level', () => {
+    const code = 'contract A { }'
+
+    const warnReport = linter.processStr(code, {
+      rules: { 'no-empty-blocks': 'warn' },
+    })
+    assertWarnsCount(warnReport, 1)
+
+    const errorReport = linter.processStr(code, {
+      rules: { 'no-empty-blocks': 'error' },
+    })
+    assertErrorCount(errorReport, 1)
+  })
+
+  describe('Abstract contract cases', () => {
+    it('should not raise warning for abstract contract with virtual functions', () => {
+      const code = `
+        abstract contract Base {
+          function foo() public virtual;
+        }
+      `
+      const report = linter.processStr(code, {
+        rules: { 'no-empty-blocks': 'warn' },
+      })
+      assertNoWarnings(report)
+    })
+  })
+
+  describe('Try-catch variations', () => {
+    const TRY_CATCH_CASES = [
+      funcWith('try foo() { revert(); } catch Error(string memory) { }'),
+      funcWith('try foo() { revert(); } catch (bytes memory) { }'),
+      funcWith('try foo() returns (uint) { } catch Error(string memory) { revert(); }'),
+    ]
+
+    TRY_CATCH_CASES.forEach((code) => {
+      it(`should detect empty blocks in try-catch: ${label(code)}`, () => {
+        const report = linter.processStr(code, {
+          rules: { 'no-empty-blocks': 'warn' },
+        })
+        assertWarnsCount(report, 1)
+      })
+    })
+  })
+
+  describe('Constructor inheritance', () => {
+    it('should not raise warning for empty constructor with multiple inheritance', () => {
+      const code = `
+        contract C is A, B {
+          constructor() A(1) B(2) { }
+        }
+      `
+      const report = linter.processStr(code, {
+        rules: { 'no-empty-blocks': 'warn' },
+      })
+      assertNoWarnings(report)
+    })
+  })
+
+  describe('Modifier cases', () => {
+    it('should raise warning for empty modifier', () => {
+      const code = contractWith('modifier onlyOwner() { }')
+      const report = linter.processStr(code, {
+        rules: { 'no-empty-blocks': 'warn' },
+      })
+      assertWarnsCount(report, 1)
+    })
+
+    it('should not raise warning for modifier with placeholder', () => {
```

Comment:
we should report on this unless the `allowEmptyModifiers` option is set to `true`. This is the normal 'empty modifiers' case, the one above is forbidden by the compiler. I encourage you to init a small foundry project and try this kind of things out, might make your life a lot easier :grin: